src/Symfony/Component/Security/Core/Exception/UserNotFoundException.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Exception;
+
+/**
+ * UserNotFoundException is thrown if a User cannot be found for the given identifier.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Alexander <iam.asm89@gmail.com>
+ */
+class UserNotFoundException extends AuthenticationException
+{
+    private $identifier;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMessageKey()
+    {
+        return 'Username could not be found.';
```

Comment:
How would we consider BC on this public error message? I think we cannot change it?

Reply:
No, it's not  BC break. We can change it, but (not a blocker) that invalidates all translations (so, we need updates on translations as well).

Reply:
that also break things if a project was customizing the translation messages in their own messages, as they won't be overriding the translation being used anymore.

Reply:
After @stof's comment, I think it's best to keep the message unchanged.

Also, `User identifier could not be found` is a much to technical error message to have as a default user-friendly message imho. I think most applications that care already updated this to say e.g. "E-mailaddress could not be found", so "Username" seems a good default for the others.